lighthouse-core/report/html/renderer/category-renderer.js

```diff
@@ -356,6 +352,29 @@ class CategoryRenderer {
     return tmpl;
   }
 
+  /**
+   * Define the score arc of the gauge
+   * Credit to xgad for the original technique: https://codepen.io/xgad/post/svg-radial-progress-meters
+   * @param {SVGCircleElement} elem
+   * @param {number} percent
+   */
+  _setGaugeArc(elem, percent) {
+    const circumferencePx = 2 * Math.PI * Number(elem.getAttribute('r'));
+    // The rounded linecap of the stroke extends the arc past its start & end.
+    // First, we tweak the -90deg rotation to adjust
+    const strokeWidthPx = Number(elem.getAttribute('stroke-width'));
+    const rotationalAdjustmentPercent = 0.25 * strokeWidthPx / circumferencePx;
+    elem.style.transform = `rotate(${-90 + rotationalAdjustmentPercent * 360}deg)`;
+
+    // Then, we terminate the line a little early as well.
+    let arcLengthPx = percent * circumferencePx - strokeWidthPx / 2;
+    // Special cases. No dot for 0, and full ring if 100
+    if (percent === 0) elem.style.opacity = '0';
+    if (percent === 1) arcLengthPx += strokeWidthPx / 2;
```

Comment:
this might be clearer as 
```suggestion
    if (percent === 1) arcLengthPx = circumferencePx;
```